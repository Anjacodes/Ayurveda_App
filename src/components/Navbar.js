import {
  Link,
} from 'react-router-dom';
import styles from './Navbar.module.css';

const Navbar = () => (
  <nav className={styles.navCont}>
    <h1 className={styles.title}><Link className={styles.link} to="/">Math Magicians</Link></h1>
    <ul className={styles.linkCont}>
      <li className={styles.linkEl}>
        <Link className={styles.link} to="/">Home</Link>
      </li>
      <li className={styles.linkEl}>
        <Link className={styles.link} to="/calculator">Calculator</Link>
      </li>
      <li className={styles.linkEl}>
        <Link className={styles.link} to="/quote">Quote</Link>
      </li>
    </ul>
  </nav>
);

export default Navbar;
