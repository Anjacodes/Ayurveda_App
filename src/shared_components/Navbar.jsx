import React, { useState } from 'react';

import {
  Link,
} from 'react-router-dom';
import NavPanel from './NavPanel';

const Navbar = () => {
  const [menuVisible, setMenuVisible] = useState(false);

  return (
    <nav className="flex flex-col items-center">
      <NavPanel visible={menuVisible} />
      <div className="flex p-4 fixed z-10 bg-slate-100 mt-8 w-[90vw] rounded-md justify-evenly">
        <div className="w-[55%] flex justify-between">
          {menuVisible ? <span onClick={() => setMenuVisible(false)}><i className="fa-solid fa-x hover:cursor-pointer" /></span> : <span onClick={() => setMenuVisible(true)}><i className="text-2xl fa-solid fa-bars hover:cursor-pointer" /></span>}
          <h1 className="font-elanor text-2xl"><Link to="/">Ayurhealth</Link></h1>
        </div>
        <ul className="w-[45%] flex justify-end items-center">
          <li>
            <Link to="/">Login</Link>
          </li>
          <li className="ml-2">
            <Link to="/calculator">Account</Link>
          </li>
        </ul>
      </div>
    </nav>
  );
};

export default Navbar;
