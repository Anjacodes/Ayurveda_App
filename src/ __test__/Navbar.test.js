import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import Navbar from '../components/Navbar';

it('renders the navbar correctly', () => {
  const tree = renderer.create(<Router><Navbar /></Router>);
  expect(tree).toMatchSnapshot();
});
