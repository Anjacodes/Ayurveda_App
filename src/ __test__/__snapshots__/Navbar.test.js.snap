// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the navbar correctly 1`] = `
<nav
  className="navCont"
>
  <h1
    className="title"
  >
    <a
      className="link"
      href="/"
      onClick={[Function]}
    >
      Math Magicians
    </a>
  </h1>
  <ul
    className="linkCont"
  >
    <li
      className="linkEl"
    >
      <a
        className="link"
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
    </li>
    <li
      className="linkEl"
    >
      <a
        className="link"
        href="/calculator"
        onClick={[Function]}
      >
        Calculator
      </a>
    </li>
    <li
      className="linkEl"
    >
      <a
        className="link"
        href="/quote"
        onClick={[Function]}
      >
        Quote
      </a>
    </li>
  </ul>
</nav>
`;
